packages/google_maps_flutter/lib/src/google_map.dart

```diff
@@ -58,9 +68,24 @@ class GoogleMap extends StatefulWidget {
   /// True if the map view should relay camera move events to Flutter.
   final bool trackCameraPosition;
 
-  // Markers to be placed on the map.
+  /// Markers to be placed on the map.
   final Set<Marker> markers;
 
+  /// Callback for when the camera move started.
+  final VoidCallback onCameraMoveStarted;
+
+  /// Callback for when the camera is moving.
+  final CameraPositionCallback onCameraMove;
+
+  /// Callback for when the camera has entered an idle stage.
+  final VoidCallback onCameraIdle;
+
+  /// Callback when the [GoogleMap] options have been applied on the platform.
+  final CameraPositionCallback onMapOptionsUpdate;
```

Comment:
I wonder if the onOptionsUpdate and onMarkersUpdate callback are really needed? do you have some concrete use-cases in mind?
We can always add fields later, but removing will be a breaking API change.

Reply:
They provide a way for users to know when the platform has "committed" the update. It might be useful to take an action based on it. We could remove them, I put them here to maintain compatibility with the behavior before.